dts/bindings/base/base.yaml

```diff
@@ -17,78 +26,169 @@ properties:
+      "interrupt-parent" will use the same format in their interrupts properties.
+
+      For details, see "2.4 Interrupts and Interrupt Mapping" in
+      Devicetree Specification v0.4.
 
   # Does not follow the 'type: phandle-array' scheme, but gets type-checked
   # by the code. Declare it here just so that other bindings can make it
   # 'required: true' easily if they want to.
   interrupts-extended:
     type: compound
-    description: extended interrupt specifier for device
+    description: |
+      Extended interrupt specifier for device, used as an alternative to
+      the "interrupts" property.
+
+      For details, see "2.4 Interrupts and Interrupt Mapping" in
+      Devicetree Specification v0.4.
 
   interrupt-names:
     type: string-array
-    description: name of each interrupt
+    description: |
+      Optional names given to each interrupt generated by a device.
+      The interrupts themselves are defined in either "interrupts" or
+      "interrupts-extended" properties.
+
+      For details, see "2.4 Interrupts and Interrupt Mapping" in
+      Devicetree Specification v0.4.
 
   interrupt-parent:
     type: phandle
-    description: phandle to interrupt controller node
+    description: |
+      If present, this refers to the node which handles interrupts generated
+      by this device.
+
+      For details, see "2.4 Interrupts and Interrupt Mapping" in
+      Devicetree Specification v0.4.
 
   label:
     type: string
     deprecated: true
     description: |
-      Human readable string describing the device (used as device_get_binding() argument)
+      Human readable string describing the device. Use of this property is
+      deprecated except as needed on a case-by-case basis.
+
+      For details, see "4.1.2 Miscellaneous Properties" in Devicetree
+      Specification v0.4.
 
   clocks:
     type: phandle-array
-    description: Clock gate information
+    description: |
+      Information about the device's clock providers. In general, this property
+      should follow conventions established in the dt-schema binding:
+
+        https://github.com/devicetree-org/dt-schema/blob/main/dtschema/schemas/clock/clock.yaml
 
   clock-names:
     type: string-array
-    description: name of each clock
+    description: |
+      Optional names given to each clock provider in the "clocks" property.
 
   "#address-cells":
     type: int
-    description: number of address cells in reg property
+    description: |
+      This property encodes the number of bits required by register block
+      base addresses in "reg" properties in this node's children.
+
+      For details, see "2.3.5 #address-cells and #size-cells" in Devicetree
+      Specification v0.4.
 
   "#size-cells":
     type: int
-    description: number of size cells in reg property
+    description: |
+      This property encodes the number of bits required by register block
```

Comment:
Technically, it encodes the number of u32 _cells_ required by register block sizes. As far as I know, there is no established convention that these must be concatenated in order to form a total number of bits (i.e. `#size-cells = <2>` meaning 64 bit size)?

Reply:
I think we should reword both the `#size-cells` and `#address-cells` descriptions to be more like what is in the devicetree spec 0.4, section 2.3.5:

> The #address-cells property defines the number of \<u32\> cells used to encode the address field in a childnode’s reg property. The #size-cells property defines the number of \<u32\> cells used to encode the size field in a childnode’s reg property.